app/src/main/java/io/appium/uiautomator2/core/EventRegister.java

```diff
@@ -2,46 +2,94 @@
 
 
 import android.app.UiAutomation;
-import android.support.test.uiautomator.Configurator;
 import android.view.accessibility.AccessibilityEvent;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeoutException;
 
 import io.appium.uiautomator2.model.AccessibilityScrollData;
 import io.appium.uiautomator2.model.AppiumUiAutomatorDriver;
+import io.appium.uiautomator2.model.NotificationListener;
 import io.appium.uiautomator2.model.Session;
 import io.appium.uiautomator2.utils.Logger;
 
 public abstract class EventRegister {
 
+    public static int EVENT_COOLDOWN_MS = 750;
```

Comment:
This timeout should be configurable, because it adds almost one second to all commands wrapped by `EventRegister.runAndRegisterScrollEvents` (like `touch[Up|Down|Move]`, `swipe`) if execution of this commands did not generate scroll event.

Reply:
@vmaxim configurable as a capability, probably?

Reply:
Via settings  - it should be available at runtime. You can use buildin timeout [setScrollAcknowledgmentTimeout](https://developer.android.com/reference/android/support/test/uiautomator/Configurator.html#setScrollAcknowledgmentTimeout(long)).

Reply:
@vmaxim ok what do you think about this

Reply:
oh, just saw your comment n/m

Reply:
@vmaxim i'm not sure settings is currently set up to work well with this use case. we don't actually want to call `setScrollAcknowledgementTimeout` i don't think, because we have to explicitly pass a `timeout` value into `executeAndWaitForEvent`. So we will want to retrieve the value from the setting every time we use this. Currently `AbstractSetting` isn't set up to save values, nor is there any kind of singleton thing going on with the actual settings classes, to ensure we can get the value back out.

i could do all of that refactoring, or just keep with the capability for now. what do you think?

Reply:
OK,  let's keep it in d-caps, so users which don't care about scroll data can set it to zero and disable all this stuff

Reply:
Sounds good. This should be a pretty niche feature. Does the code as it stands look good to you then?